src/model/enhance.ts

```diff
@@ -0,0 +1,62 @@
+import { JSDOM } from 'jsdom';
+
+const TEXT = 'model.dotcomrendering.pageElements.TextBlockElement';
+const SUBHEADING = 'model.dotcomrendering.pageElements.SubheadingBlockElement';
+
+const isDropCapFlag = (element: SubheadingBlockElement): boolean => {
+    // A drop cap flag: <h2><strong>* * *</strong></h2>
+    const frag = JSDOM.fragment(element.html);
+    if (!frag || !frag.firstChild) return false;
+    return frag.firstChild.nodeName === 'H2' && frag.textContent === '* * *';
+};
+
+const enhanceElements = (elements: CAPIElement[]): CAPIElement[] => {
+    // enhanceElements loops the array of article elements making 'enhancements' (read hacks)
+    // to allow us to support the data model coming out of CAPI. Similar logic exists in frontend
+    // here: https://github.com/guardian/frontend/blob/08b3732d113c28c5165ab2ce8e9b7b52909da336/common/app/views/support/HtmlCleaner.scala
+    // TODO: Take these changes and push them up the stack so that we don't need to make logic
+    // overrides in the rendering tier.
+    //
+    // The current list of enhancements are:
+    //
+    // 1. Drop Cap
+    //    If a h2 tag containing * * * is found, then the proceeding element should be givin drop
+    //    cap styling (so long as it's a text element).
+    //
+    let dropCapFlagSeen = false;
+
+    const enhanced: CAPIElement[] = [];
+    elements.forEach(element => {
```

Comment:
Hey, so I quickly tried to write a spike of what I imagined of enhance CAPI (it's not working, so assume it pseudo code). 

The main difference I had in mind was that rather than tracking through state, each 'block' has an enhance method that take the previous block and current and self contains the updating. Within that I imagined self contained methods to do the rewriting.

It's very quickly written and rough but wondered on your thoughts as an alternative?

https://gist.github.com/gtrufitt/1dc327066dbfbfa34a00ed61bd610549

Reply:
I actually think I still like the 'sanitize-html' approach as a way to better define an API for handling this and avoiding dom soup 

https://github.com/guardian/dotcom-rendering/blob/2616d4d30bfaed89cbdc3acd8d119c357aa6e824/src/web/components/elements/TextBlockComponent.tsx#L41

But obviously everything can't be handled with it

Reply:
This is really nice:

```
let shouldDropCap = prev.type === 'subheadingBlock' && prev.content.includes('***');
```

I worry it might look less nice if the same logic needs to be aware of more than just the previous element though, like with photo essays

But if we take the one enhancer file per use case approach, then we can implement each one as we like. Only having to write code to deal with one problem is going to keep things a lot easier to reason about and also make removing it later easier too

Reply:
Also, again, assuming we keep processing separate, then perhaps we could use the `sanitzer` on a case by case bases? It seems like it would work well for edits in place without the  need for context